client/plots/brainImaging.js

```diff
@@ -247,153 +223,152 @@ class BrainImaging {
-			const data = await dofetch3('brainImaging', { body })
-			this.legendValues = data.legend
-			if (data.error) throw data.error
 
-			this.dataUrlL = {}
-			for (const [termV, result] of Object.entries(data.brainImage)) {
-				this.dataUrlL[termV] = result
-			}
+		//update L image //////////////////
+		let body = {
+			genome: this.state.genome,
+			dslabel: this.state.dslabel,
+			refKey: this.state.config.queryKey,
+			l: this.settings.brainImageL,
+			selectedSampleFileNames: this.state.config.selectedSampleFileNames,
+			divideByTW,
+			overlayTW,
+			legendFilter: this.state.config.legendFilter
+		}
+		let data = await dofetch3('brainImaging', { body })
+		this.legendValues = data.legend
+		if (data.error) throw data.error
 
-			this.dom.tdL.selectAll('*').remove()
-			this.dom.imagesL = []
-			for (const [termV, result] of Object.entries(this.dataUrlL)) {
-				if (divideByTW)
-					this.dom.tdL
-						.append('div')
-						.attr('class', 'pp-chart-title')
-						.style('text-align', 'center')
-						.text(`${termV} (n=${result.catNum})`)
-						.style('font-weight', '600')
-						.style('color', 'white')
-						.style('font-size', '24px')
-						.style('margin-bottom', '5px')
-						.style('margin-top', '5px')
-						.style('display', 'block')
-				const img = this.dom.tdL.append('div').append('img').attr('src', result.url)
-				this.dom.imagesL.push(img)
-			}
+		this.dataUrlL = {}
+		for (const [termV, result] of Object.entries(data.brainImage)) {
+			this.dataUrlL[termV] = result
 		}
 
-		if (this.state.updateF) {
-			const body = {
-				genome: this.state.genome,
-				dslabel: this.state.dslabel,
-				refKey: this.state.config.queryKey,
-				f: this.settings.brainImageF,
-				selectedSampleFileNames: this.state.config.selectedSampleFileNames,
-				divideByTW,
-				overlayTW,
-				legendFilter: this.state.config.legendFilter
-			}
-			const data = await dofetch3('brainImaging', { body })
-			if (data.error) throw data.error
+		this.dom.tdL.selectAll('*').remove()
+		this.dom.imagesL = []
+		for (const [termV, result] of Object.entries(this.dataUrlL)) {
+			if (divideByTW)
+				this.dom.tdL
+					.append('div')
+					.attr('class', 'pp-chart-title')
+					.style('text-align', 'center')
+					.text(`${termV} (n=${result.catNum})`)
+					.style('font-weight', '600')
+					.style('color', 'white')
+					.style('font-size', '24px')
+					.style('margin-bottom', '5px')
+					.style('margin-top', '5px')
+					.style('display', 'block')
+			const img = this.dom.tdL.append('div').append('img').attr('src', result.url)
+			this.dom.imagesL.push(img)
+		}
+		//update F image //////////////////
+		body = {
+			genome: this.state.genome,
+			dslabel: this.state.dslabel,
+			refKey: this.state.config.queryKey,
+			f: this.settings.brainImageF,
+			selectedSampleFileNames: this.state.config.selectedSampleFileNames,
+			divideByTW,
+			overlayTW,
+			legendFilter: this.state.config.legendFilter
+		}
+		data = await dofetch3('brainImaging', { body })
```

Comment:
reusing `body` and `data` variable names is a bit risky, it would be better to reuse a separate function for the different image requests, so that body and data could be constant inside the function. If done this way, could also parallelize the 3 requests like this with `Promise.all(requests)`.